File: crudJavaDocker/crud/src/main/java/com/crudApi/crud/controller/PersonController.java
package com.crudApi.crud.controller;

import com.crudApi.crud.model.Person;
import com.crudApi.crud.repository.PersonRepository;
import org.hibernate.mapping.Value;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("persons")
@CrossOrigin("*")
public class PersonController {

    private final PersonRepository personRepository;

    public PersonController(PersonRepository personRepository) {
        this.personRepository = personRepository;
    }

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public List<Person> Persons() {return personRepository.findAll();}

    @GetMapping(path = {"/{id}"})
    public ResponseEntity findById(@PathVariable long id) {
        return personRepository.findById(id)
                .map(person1 -> ResponseEntity.ok().body(person1))
                .orElse(ResponseEntity.notFound().build());
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Person CustomerCreate(@RequestBody Person person) {
        personRepository.save(person);
        return person;
    }
    @PutMapping(value = "/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public ResponseEntity<Person> Update(@PathVariable("id") long id, @RequestBody Person person) {
        return personRepository.findById(id).map(
                person1 -> {
                    person1.setName(person.getName());
                    person1.setCity((person.getCity()));
                    Person atualizado = personRepository.save(person1);
                    return ResponseEntity.ok().body(atualizado);
                }).orElse(ResponseEntity.notFound().build());
    }
    @DeleteMapping(path = {"/{id}"})
    public ResponseEntity<?> delete(@PathVariable long id) {
        return personRepository.findById(id)
                .map(person1 -> {
                            personRepository.deleteById(id);
                            return ResponseEntity.ok().build();
                        }).orElse(ResponseEntity.notFound().build());
    }


}

File: kocru/src/main/kotlin/com/crud/kotlin/kocru/controller/PersonController.kt
package com.crud.kotlin.kocru.controller

import com.crud.kotlin.kocru.controller.request.PostPersonRequest
import com.crud.kotlin.kocru.controller.request.PutPersonRequest
import com.crud.kotlin.kocru.controller.response.PeopleResponse
import com.crud.kotlin.kocru.extension.toPersonModel
import com.crud.kotlin.kocru.extension.toResponse
import com.crud.kotlin.kocru.repository.PersonRepository
import com.crud.kotlin.kocru.service.PersonService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("persons")
@CrossOrigin("*")
class PersonController(private val personService: PersonService, private val personRepository: PersonRepository) {

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    fun getPerson(@PathVariable id: Int): PeopleResponse {
        return personService.findById(id).toResponse()
    }

    @GetMapping()
    @ResponseStatus(HttpStatus.OK)
    fun getAll():List<PeopleResponse> {
        return personService.getAll().map {it.toResponse()}
    }
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun create(@RequestBody person: PostPersonRequest) {
        personRepository.save(person.toPersonModel())
    }
    @PutMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun update(@PathVariable id: Int, @RequestBody person: PutPersonRequest) {
        val personSaved = personService.findById(id)
        return personService.update(person.toPersonModel(personSaved))
    }
    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun delete(@PathVariable id:Int) {
        personService.delete(id)
    }
}